Makefile.cache

```diff
@@ -0,0 +1,637 @@
+endef
+
+
+# Calculate the 24 byte SHA value
+# SHA value is derived from dependent files of the target which includes .flags, .sha and .smsha files.
+# Args:
+#  $(1)   => target name
+define GET_MOD_DEP_SHA
+	$(eval $(1)_MOD_DEP_PKGS := $(foreach dfile,$($(1)_DEPENDS) $($(1)_RDEPENDS) $($(1)_WHEEL_DEPENDS) \
+                            $($(1)_PYTHON_DEBS) $($(1)_PYTHON_WHEELS) \
+                            $($(1)_DBG_DEPENDS) $($(1)_DBG_IMAGE_PACKAGES) $($(1)_LOAD_DOCKERS),\
+                            $(if $($(dfile)_MAIN_DEB),$($(dfile)_MAIN_DEB),$(dfile))) )
+
+	$(if $(MDEBUG), $(info DEP_FILES: $($(1)_MOD_DEP_PKGS)))
+	$(eval $(1)_DEP_MOD_SHA := $(shell git hash-object $($(1)_DEP_FLAGS_FILE) \
+                            $(foreach dfile,$($(1)_MOD_DEP_PKGS),$($(dfile)_MOD_HASH_FILE) $($(dfile)_SMOD_HASH_FILE) ) \
+                            | sha1sum | awk '{print substr($$1,0,23);}'))
+endef
+
+
+# Retrive the list of files that are modified for the target. The files can be from
+# 1. Any of dependent target is modified
+# 2. Files from the target dependency list
+# 3. Files from submodule dependency list if the target is a submodule
+#
+# Args:
+#  $(1)   => target name
+define GET_MODIFIED_FILES
+	$(eval $(1)_FILES_MODIFIED := $(foreach dfile,$($(1)_MOD_DEP_PKGS),$(if $($(dfile)_FILES_MODIFIED),$(dfile))) \
+                                  $(if $($(1)_DEP_FILES), $(shell cat $($(1)_MOD_DEP_FILE) | xargs git status -s)) \
+                                  $(if $($(1)_SMDEP_PATHS), $(foreach path,$($(1)_SMDEP_PATHS), \
+                                        $(shell cd $(path) &&  git status --ignore-submodules=all -s -uno .))) )
+
+endef
+
+# Loads the deb package from debian cache
+# Cache file prefix is formed using SHA value
+# The SHA value consists of
+#   1.  24 byte SHA value is derived from the dependency files list
+#          Flags: Module ENV flags file
+#          Dependent packages : DEPENDS, RDEPENDS, WHEEL_DEPENDS, PYTHON_DEBS, PYTHON_WHEELS, DBG_DEPENDS, DBG_IMAGE_PACKAGES, LOAD_DOCKERS
+#   2.  24 byte SHA value from one of the keyword type - GIT_COMMIT_SHA or GIT_CONTENT_SHA
+#          GIT_COMMIT_SHA   - SHA value of the last git commit id if it is a submodule
+#          GIT_CONTENT_SHA  - SHA value is calculated from the target dependency files content.
+#   Cache is loaded from either local cache or global cache based on the dependency SHA match with the cache filename.
+#   Otherwise it builds the package from source.
+# TODO:
+# 1. Extend dpkg for all the chip vendor packages.
+
+# Args:
+# $(1) => target name
+# $(2) => target output file name
+define LOAD_FROM_CACHE
+
+	# Calculate the modules SHA and its dependency SHA value
+	$(call GET_MOD_DEP_SHA,$(1))
+	$(call GET_MOD_SHA,$(1))
+
+	# Form the cache file name
+    $(eval $(1)_MOD_CACHE_FILE      := $(1)-$($(1)_DEP_MOD_SHA)-$($(1)_MOD_HASH).tgz)
```

Comment:
please use tab here, the spacing here is not correct

Reply:
Fixed